validator.js

```diff
@@ -468,7 +468,37 @@
     };
 
     validator.isDate = function (str) {
-        return !isNaN(Date.parse(str));
+        var normalizedDate = new Date((new Date(str)).toUTCString());
+        var regularDay = String(normalizedDate.getDate());
+        var utcDay = String(normalizedDate.getUTCDate());
+        var dayOrYear, dayOrYearMatches, year;
+        if (isNaN(Date.parse(normalizedDate))) {
+            return false;
+        }
+        if (typeof str !== 'string') {
+          return true;  //if Date.parse() passes, it's the current time
```

Comment:
This isn't necessary. The first argument is always coerced to a string ([1](https://github.com/chriso/validator.js/blob/fc03288cde9f3edd5eef2018c72262ba1153d06e/validator.js#L786), [2](https://github.com/chriso/validator.js/blob/fc03288cde9f3edd5eef2018c72262ba1153d06e/validator.js#L107-L117), [3](https://github.com/chriso/validator.js/blob/fc03288cde9f3edd5eef2018c72262ba1153d06e/validator.js#L99-L105)) based on these rules: https://github.com/chriso/validator.js#strings-only